patches/node/add_openssl_is_boringssl_guard_to_oaep_hash_check.patch

```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shelley Vohr <shelley.vohr@gmail.com>
+Date: Wed, 21 Aug 2019 12:43:26 -0700
+Subject: Add OPENSSL_IS_BORINGSSL guard to oaep_hash check
+
+Adds an ifndef guard to the oaep_hash nullptr check in
+PublicKeyCipher::Cipher, since BORINGSSL does not expose either
+EVP_PKEY_OP_TYPE_CRYPT or EVP_PKEY_CTRL_RSA_OAEP_MD.
+
+diff --git a/src/node_crypto.cc b/src/node_crypto.cc
+index 59b60f1aea11666a7393831f8a465c247b66b009..1b44572c3a126c1ba3f610dbefb226acb34d246d 100644
+--- a/src/node_crypto.cc
++++ b/src/node_crypto.cc
+@@ -5156,11 +5156,15 @@ bool PublicKeyCipher::Cipher(Environment* env,
+   if (EVP_PKEY_CTX_set_rsa_padding(ctx.get(), padding) <= 0)
+     return false;
+ 
++#ifdef OPENSSL_IS_BORINGSSL
```

Comment:
We do `#ifdef 0` in some other node patches for this. Not for this change, but we should update those patches also, I think.

Reply:
@nitsakh would you prefer i update those in this PR?

Reply:
Cool, thanks for doing this @codebytere . 👍